src/composables/useBarDrag.ts

```diff
@@ -1,108 +1,123 @@
-import useDayjsHelper from "./useDayjsHelper"
+import type { GGanttChartConfig } from "../components/GGanttChart.vue"
+import provideConfig from "../provider/provideConfig"
+import { type Ref, ref } from "vue"
 
+import type { GanttBarObject } from "../types"
+import useDayjsHelper from "./useDayjsHelper"
 import useTimePositionMapping from "./useTimePositionMapping"
-import { Ref, ref } from "vue"
 
-export default function useBarDrag (
+export default function createBarDrag(
   bar: Ref<GanttBarObject>,
-  gGanttChartPropsRefs: GGanttChartPropsRefs,
   onDrag: (e: MouseEvent, bar: GanttBarObject) => void = () => null,
-  onEndDrag: (e: MouseEvent, bar: GanttBarObject) => void = () => null
+  onEndDrag: (e: MouseEvent, bar: GanttBarObject) => void = () => null,
+  config: GGanttChartConfig = provideConfig()
 ) {
-  const { barStart, barEnd, pushOnOverlap } = gGanttChartPropsRefs
+  const { barStart, barEnd, pushOnOverlap } = config
 
   const isDragging = ref(false)
   let cursorOffsetX = 0
-  let dragCallBack : (e: MouseEvent) => void
+  let dragCallBack: (e: MouseEvent) => void
 
-  const { mapPositionToTime } = useTimePositionMapping(gGanttChartPropsRefs)
-  const { toDayjs } = useDayjsHelper(gGanttChartPropsRefs)
+  const { mapPositionToTime } = useTimePositionMapping(config)
+  const { toDayjs } = useDayjsHelper(config)
 
   const initDrag = (e: MouseEvent) => {
     const barElement = document.getElementById(bar.value.ganttBarConfig.id)
-    if (barElement) {
-      cursorOffsetX = e.clientX - (barElement.getBoundingClientRect().left || 0)
-      const mousedownType = (e.target as Element).className
-      switch (mousedownType) {
-        case "g-gantt-bar-handle-left":
-          document.body.style.cursor = "w-resize"
-          dragCallBack = dragByLeftHandle
-          break
-        case "g-gantt-bar-handle-right":
-          document.body.style.cursor = "w-resize"
-          dragCallBack = dragByRightHandle
-          break
-        default: dragCallBack = drag
-      }
-      isDragging.value = true
-      window.addEventListener("mousemove", dragCallBack)
-      window.addEventListener("mouseup", endDrag)
+    if (!barElement) {
+      return
+    }
+
+    cursorOffsetX = e.clientX - (barElement.getBoundingClientRect().left || 0)
+    const mousedownType = (e.target as Element).className
+    switch (mousedownType) {
+      case "g-gantt-bar-handle-left":
+        document.body.style.cursor = "ew-resize"
+        dragCallBack = dragByLeftHandle
+        break
+      case "g-gantt-bar-handle-right":
+        document.body.style.cursor = "ew-resize"
+        dragCallBack = dragByRightHandle
+        break
+      default:
+        dragCallBack = drag
     }
+    isDragging.value = true
+    window.addEventListener("mousemove", dragCallBack)
+    window.addEventListener("mouseup", endDrag)
   }
 
-  const drag = (e: MouseEvent) => {
+  function getBarElements() {
```

Comment:
Good idea to wrap this in a helper function! How about converting in to an arrow function so that it's consistent with other functions?

Reply:
Fixed in 26c13c64d4a2a13e07ea1780924834ce3b77bc90